app/scripts/modules/core/src/entityTag/EntityTagEditor.tsx

```diff
@@ -0,0 +1,322 @@
+import { IDeferred } from 'angular';
+import { $q } from 'ngimport';
+import * as Formsy from 'formsy-react';
+import * as marked from 'marked';
+import * as React from 'react';
+import { Modal } from 'react-bootstrap';
+import { IModalServiceInstance } from 'angular-ui-bootstrap';
+import autoBindMethods from 'class-autobind-decorator';
+
+import {
+  UUIDGenerator, Application, EntityTagWriter, HelpField, IEntityRef, IEntityTag,
+  ReactInjector, TaskMonitor, TaskMonitorBuilder
+} from 'core';
+
+import { BasicFieldLayout, TextArea, ReactModal } from 'core/presentation';
+import { NgReact } from 'core/reactShims/ngReact';
+import { EntityRefBuilder } from './entityRef.builder';
+
+import './entityTagEditor.modal.less';
+
+export interface IOwnerOption {
+  label: string;
+  type: string;
+  owner: any;
+  isDefault: boolean;
+}
+
+export interface IEntityTagEditorProps {
+  owner: any;
+  application: Application;
+  entityType: string;
+  tag: IEntityTag;
+  ownerOptions: IOwnerOption[];
+  entityRef: IEntityRef;
+  isNew: boolean;
+  show?: boolean;
+  onHide?(event: any): void;
+  onUpdate?(): any;
+}
+
+export interface IEntityTagEditorState {
+  taskMonitor: TaskMonitor;
+  message: string;
+  show: boolean;
+  isValid: boolean;
+  isSubmitting: boolean;
+  owner: any;
+  entityType: string;
+}
+
+@autoBindMethods
+export class EntityTagEditor extends React.Component<IEntityTagEditorProps, IEntityTagEditorState> {
+  public static defaultProps = {
+    onHide: () => null as any,
+    onUpdate: () => null as any,
+  };
+
+  private taskMonitorBuilder: TaskMonitorBuilder = ReactInjector.taskMonitorBuilder;
+  private entityTagWriter: EntityTagWriter = ReactInjector.entityTagWriter;
+  private $uibModalInstanceEmulation: IModalServiceInstance & { deferred?: IDeferred<any> };
+
+  /** Shows the Entity Tag Editor modal */
+  public static show(props: IEntityTagEditorProps): Promise<any> {
```

Comment:
In the generic sense, it (`Modal.show`) actually does resolve to the "onHide action".  This could be used to return a promise for a radio choice in a modal, for example.

In this case (EntityTagEditor) it doesn't resolve to anything, so changing to void is proper